src/actors/objects/mod.rs

```diff
@@ -555,60 +559,73 @@ impl ObjectsActor {
+                        .or_else(move |e| {
+                            // This basically only happens for the Sentry source type, when doing the
+                            // search by debug/code id. We do not surface those errors to the user
+                            // (instead we default to an empty search result) and only report them
+                            // internally.
+                            log::error!("Failed to download from {}: {}", type_name, LogError(&e));
+                            Ok(Vec::new())
+                        })
+                        // create a new hub for each prepare_downloads call
+                        .sentry_hub_new_from_current()
+                })
+                // collect into intermediate vector because of borrowing problems
+                .collect::<Vec<_>>(),
+        )
+        .map(|file_ids: Vec<Vec<FileId>>| -> Vec<FileId> {
+            file_ids.into_iter().flatten().collect()
+        });
 
-        future::join_all(prepare_futures).and_then(move |responses| {
+        let meta_cache = self.meta_cache.clone();
+        let threadpool = self.threadpool.clone();
+        let data_cache = self.data_cache.clone();
+
+        let file_metas = file_ids.and_then(move |file_ids| {
+            future::join_all(file_ids.into_iter().map(move |file_id| {
+                let request = FetchFileMetaRequest {
+                    scope: if file_id.source().is_public() {
+                        Scope::Global
+                    } else {
+                        scope.clone()
+                    },
+                    file_id,
+                    object_id: identifier.clone(),
+                    threadpool: threadpool.clone(),
+                    data_cache: data_cache.clone(),
+                };
+
+                meta_cache
+                    .compute_memoized(request.clone())
+                    .map_err(|e| ObjectError::from(ArcFail(e).context(ObjectErrorKind::Caching)))
+                    // Errors from a file download should not make the entire join_all fail. We
+                    // collect a Vec<Result> and surface the original error to the user only when
+                    // we have no successful downloads.
+                    .then(Ok)
+                    // create a new hub for each file download
+                    .sentry_hub_new_from_current()
+            }))
+        });
+
+        file_metas.and_then(move |responses: Vec<Result<Arc<ObjectFileMeta>, _>>| {
             responses
                 .into_iter()
-                .flatten()
                 .enumerate()
                 .min_by_key(|(i, response)| {
                     // Prefer files that contain an object over unparseable files
                     let object = match response {
                         Ok(object) => object,
-                        _ => return (3, *i),
+                        Err(e) => {
+                            log::trace!("Failed to download: {}", LogError(e));
```

Comment:
```suggestion
                            log::debug!("Failed to download: {}", LogError(e));
```

Trace won't make it into breadcrumbs iirc.